weasyl/searchtag.py

```diff
@@ -118,6 +118,50 @@ def parse_tags(text):
     return tags
 
 
+def parse_blacklist_tags(text):
+    """
+    A custom implementation of ``parse_tags()`` for the searchtag blacklist.
+    Enforces the desired characteristics of STBL tags, and allows an asterisk
+       character, whereas ``parse_tags()`` would strip asterisks.
+
+    Parameters:
+        text: The string to parse for tags
+
+    Returns:
+        tags: A set() with valid tags.
+    """
+    tags = set()
+
+    for i in _TAG_DELIMITER.split(text):
+        target = "".join([c for c in i if ord(c) < 128])
+        target = target.replace(" ", "_")
```

Comment:
I think `_TAG_DELIMITER` means there can’t be any spaces in the string at this point.